internal/controller/instaslice_controller.go

```diff
@@ -586,20 +617,73 @@ func (r *InstasliceReconciler) podMapFunc(ctx context.Context, obj client.Object
 	return requests
 }
 
-// SetupWithManager sets up the controller with the Manager.
+// Initialize Prometheus-compatible profiles metrics when the controller starts
+// Adds a background goroutine that waits for Instaslice objects.
+// Proceeds to setupWithManager(mgr) to start the reconciler
+// Does not block the controller from reconciling
+// UpdateCompatibleProfilesMetrics only updates Prometheus metrics,in-memory and do not persist in etcd
 func (r *InstasliceReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	restConfig := mgr.GetConfig()
-
 	var err error
 	r.kubeClient, err = kubernetes.NewForConfig(restConfig)
 	if err != nil {
 		return err
 	}
+	mgrAddErr := mgr.Add(manager.RunnableFunc(func(ctx context.Context) error {
+		log := logr.FromContext(ctx)
+		<-mgr.Elected() // Wait for leader election before executing
+		// Retry mechanism to wait for Instaslice objects
+		var instasliceList inferencev1alpha1.InstasliceList
+		retryErr := wait.PollUntilContextTimeout(ctx, 2*time.Second, 60*time.Second, true, func(ctx context.Context) (bool, error) {
```

Comment:
Please add a TODO: support daemonset fault tolerance

Reply:
Right now metrics is not dependent on daemonset failures as it updates metrics when controller/ daemonset is healthy.

Reply:
if daemonset fails the code still re-populates those objects, I don't see any filtering

Reply:
Please add a TODO to figure out skipping logic